### DISCUSSION TITLE
Is there a way to access incremental ChatResult.chat_history as the conversation proceeds ?

### DISCUSSION THREAD

**Discussion by prasenmukh** [microsoft/autogen#4310] [General]
Is there a way I can register functions with initiate_chat() so that I can keep receiving the incremental chat_history messages. I would need incremental OpenAI responses to show it on my custom UI.  One way is to set my own stdin and stdout and then try to parse stdout. Is there a better way around so that I don't have to write a stdout parser.

**Comment by ekzhu** [comment]
This is addressed in v0.4 AgentChat. It produces a stream of messages as an async generator.

**Reply by prasenmukh** [reply]
Thanks a lot. @ekzhu Can you please refer to any code or method  pointers. The intent is to extract user interfacing messages and show it in  a different website.

**Reply by ekzhu** [reply]
Please follow the tutorial in the documentation: https://microsoft.github.io/autogen/dev/user-guide/agentchat-user-guide/index.html, focus on the Agents and Teams sections.
